warehouse/query-core/src/main/java/datawave/query/function/TLDEquality.java

```diff
@@ -23,8 +21,20 @@ public class TLDEquality implements Equality {
      */
     @Override
     public boolean partOf(Key key, Key other) {
-        ByteSequence docCF = TLD.estimateRootPointerFromId(key.getColumnFamilyData());
-        ByteSequence otherCF = TLD.estimateRootPointerFromId(other.getColumnFamilyData());
-        return otherCF.equals(docCF);
+        ByteSequence keyCf = key.getColumnFamilyData();
+        ByteSequence otherCf = other.getColumnFamilyData();
+
+        int dotCount = 0;
+        int len = Math.min(keyCf.length(), otherCf.length());
+        for (int i = 0; i < len; i++) {
+            byte a = keyCf.byteAt(i);
+            byte b = otherCf.byteAt(i);
+            if (a != b) {
+                return false;
+            } else if (a == '.' && ++dotCount == 3) {
+                return true;
+            }
+        }
+        return true;
```

Comment:
maybe this doesn't happen, but this default return will make any comparison to an empty string true.

Reply:
return len == 0 ? false : true;

??

Reply:
but then this will make empty value comparisons to false..

Reply:
I like that solution

Reply:
existing code will also evaluate these two to be the same? a.b.c and a.b.c1
maybe each part of UID has fixed length, but comparing values up to shorter length sounds a bit dangerous to me.

Reply:
Given the structure of a uid I don't think that case is possible, but I added a test to document that behavior anyway.